src/lib/coda_base/coinbase.mli

```diff
@@ -44,7 +44,11 @@ val supply_increase : t -> Currency.Amount.t Or_error.t
 val fee_excess : t -> Currency.Fee.Signed.t Or_error.t
 
 module Gen : sig
-  val gen : t Quickcheck.Generator.t
+  (** The [coinbase_amount] parameter should match the one defined in the
+      [Genesis_constants.Constraint_constants.t] used to run the tests.
+      Do NOT set it to other values.
```

Comment:
if we have this invariant, should we instead pass in a `Genesis_constants.Constraint_constants.t`? Assuming there is no dependency cycle issue with making this happen.